### ISSUE TITLE
--correct-hashes not working as expected

### ISSUE THREAD

**Issue by ataraxus** [mayope/keycloakmigration#88]
Hey, 

as promised my second finding. While trying to find a workaround for #87 i stumbled upon an unexpected behaviour. 
--correct-hashes only, at least from my point of view, works as expected if the number of migrationfile entries stays the same. 

(maybe this is the exact usecase but i had it enabled by default)

if one assumes the following migration file: 

```yaml
includes:
   - path: v001/01_system/01_add-realm.yml
   - path: v001/01_system/02_add-standard-roles.yml
```
and adds a new migration on top, like: 

```yaml
includes:
   - path: v000/01_system/01_fix-UAP.yml
   - path: v001/01_system/01_add-realm.yml
   - path: v001/01_system/02_add-standard-roles.yml
```
the migration hashes for `01_add-realm.yml` will be corrected and it will be skipped since it was already executed. Now the fun starts: `02_add-standard-roles.yml` will be wrongfully executed and very likely crash since it was already performed. 

Another issue comes up if one removes a line from the migration file: 
```yaml
includes:
   - path: v001/01_system/01_add-realm.yml
```

Then `--correct-hashes` will cause an index out of bounds exception. This is due to the nature of this implementation: 

```kotlin
 private fun List<ChangeSet>.checkExistingHashes(changeHashes: List<String>, correctHashes: Boolean) {
        changeHashes.forEachIndexed { i, it ->
            if (get(i).hash() != it) {
                handleHashError(correctHashes, it, i)
            }
            LOG.info("Skipping migration: ${get(i).id}")
        }
    }
```

where `changeHashes` is retreived only once from KC and if the current migrationfiles number is different this is an issue. 

Maybe it should be work like this: 

```
changeHashes = retreiveChangeHashes();
for(int i: migrationfiles.size(); i>=0; i--){
  migration=migrationfiles[i];
  changeHash=changeHashes.search(migration);
  if(changeHash!=null) {
    if(changeHash.position()==i){
      break; // skip migration
    } else {
      updateChangeHashes(migration,i);
    }
  } else {
    executeMigration(migration);
    updateChangeHashes(migration,i);
  }
}
```

**Comment by klg71** [comment]
You are right with the exact use case. You can only migrate hashes if the migration itself don't change. Currently this method is index based, though we could implement it based on file names I guess.
